empress_gui.py

```diff
@@ -319,14 +319,16 @@ def close_unnecessary_windows_if_opened(self):
         if self.set_num_cluster_window is not None and self.set_num_cluster_window.winfo_exists():
             self.set_num_cluster_window.destroy()
 
-        if self.view_solution_space_window is not None and self.view_solution_space_window.winfo_exists():
-            self.view_solution_space_window.destroy()
+        if self.view_solution_space_window_after_setting_clusters is not None and self.view_solution_space_window_after_setting_clusters.winfo_exists():
+            self.view_solution_space_window_after_setting_clusters.destroy()
 
         if self.one_MPR_window is not None and self.one_MPR_window.winfo_exists():
             self.one_MPR_window.destroy()
 
-        if self.view_reconciliations_window is not None and self.view_reconciliations_window.winfo_exists():
-            self.view_reconciliations_window.destroy()
+        if self.num_cluster is not None:
+            for i in range(self.num_cluster):
```

Comment:
It seems like you are iterating from `i = 0` to `i = self.num_cluster - 1` but you never used the value `i`.

Reply:
Yes, because I wanted to create self.num_cluster number of new windows.

Reply:
Yes, because I wanted to delete the self.num_cluster number of tkinter windows created by "One per cluster".

Reply:
I'm confused about what this code is doing. It seems to me that `self.solution_window` is one variable and you are attempting to destroy the same `self.solution_window` multiple number of times.

Reply:
Actually, this is not working currently and I will remove this. This issue is mentioned in the description of this pull request.

Reply:
I have changed the code. So currently only one window created by "One per cluster" can be closed when it becomes irrelevant (e.g. loading in new input files) regardless of how many new tkinter windows created by "One per cluster". I can create an issue for this if it is necessary.

Reply:
I'm not sure what the problem is, could you create a new issue for this with a number of screenshots?